### PR TITLE
Module account, event manager

### PR DESCRIPTION
**Pull request by Stumble**
TLDR:
1. As long as you trust those several lines of code in account module are correct, coins can NEVER be created from air or burned. 
2. The total amount of coin will be changed accroding the inflation schedule.
3. Wrong events will no longer panic. Instead, errors will be logged and can be adjusted later.

This PR:
1. introduce module accounts, or Pool Coin that tracks coins belongs to modules.
2. refactor account interface to allow only MOVE coins, but not "add/sub" coins.
3. refactor global manager to an event manager.
4. handles module event errors by execute events in a isolated store, rollback upon errs. Errors will be logged and can be handled later.
5. Eliminates all non-determinisitc code by setting to a fixed time.

### CODE REVIEW

**Comment by Stumble** [comment]
Cli and queries of global module are broken as well, need to be fixed.

File: x/global/manager/manager.go [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+		daily:       daily,
+		monthly:     monthly,
+		yearly:      yearly,
+	}
+}
+
+func (gm GlobalManager) InitGenesis(ctx sdk.Context) {
+	// will be updated on the first OnBeginBlock.
+	gm.storage.SetGlobalTime(ctx, &model.GlobalTime{
+		ChainStartTime: ctx.BlockTime().Unix(),
```
[the remark this answers is not in the document]
Comment (Stumble) [reply x/global/manager/manager.go]: inited at the same time, and can never be changed, not possible.

File: x/vote/manager/manager.go [hunk 2]
```diff
@@ -56,43 +70,36 @@ func (vm VoteManager) DoesVoterExist(ctx sdk.Context, username linotypes.Account
...
 
-func (vm VoteManager) AddStake(ctx sdk.Context, username linotypes.AccountKey, amount linotypes.Coin) sdk.Error {
-	if !vm.am.DoesAccountExist(ctx, username) {
-		return types.ErrAccountNotFound()
-	}
+	return vm.addStake(ctx, receiver, amount)
+}
 
+func (vm VoteManager) addStake(ctx sdk.Context, username linotypes.AccountKey, amount linotypes.Coin) sdk.Error {
 	voter, err := vm.storage.GetVoter(ctx, username)
```
[the remark this answers is not in the document]
Comment (Stumble) [reply x/vote/manager/manager.go]: removed, will fail in MoveToPool, no need to check again.
